packages/react/src/combobox/root/ComboboxRootInternal.tsx

```diff
@@ -714,9 +714,13 @@ export function ComboboxRootInternal<Value = any, Mode extends SelectionMode = '
 
       if (multiple) {
         const currentSelectedValue = Array.isArray(selectedValue) ? selectedValue : [];
-        const isCurrentlySelected = currentSelectedValue.includes(value);
+        const isCurrentlySelected = itemIncludes(
+          currentSelectedValue,
+          value,
+          store.state.isItemEqualToValue,
+        );
         const nextValue = isCurrentlySelected
-          ? currentSelectedValue.filter((v) => v !== value)
+          ? currentSelectedValue.filter((v) => !store.state.isItemEqualToValue(v, value))
```

Comment:
The `removeItem` util is used by Select here

Reply:
Fixed